### PR TITLE
keychain: add page

### PR DESCRIPTION
**Pull request by rpdelaney**
- [x] The page (if new), does not already exist in the repository.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

**Version of the command being documented (if known):**
```console
$ keychain --version

 * keychain 2.8.5 ~ http://www.funtoo.org

   Copyright 2002-2006 Gentoo Foundation;
   Copyright 2007 Aron Griffis;
   Copyright 2009-2017 Funtoo Solutions, Inc;
   lockfile() Copyright 2009 Parallels, Inc.
```

### CODE REVIEW

File: pages/common/keychain.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# keychain
+
+> re-use ssh-agent and/or gpg-agent between logins
```
Comment (CleanMachine1) [inline pages/common/keychain.md:3]:
```suggestion
> Re-use ssh-agent and/or gpg-agent between logins
```

File: pages/common/keychain.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+- Check for a running ssh-agent, and start one if needed:
+
+`keychain`
+
+- Also check for gpg-agent:
+
+`keychain --agents "gpg,ssh"`
+
+- List signatures of all active keys and exit:
```
Comment (CleanMachine1) [inline pages/common/keychain.md:14]:
```suggestion
- List signatures of all active keys:
```
Please remove this from the other lines as well.

File: pages/common/keychain.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+- List fingerprints of all active keys and exit:
+
+`keychain --list-fp`
+
+- Add a timeout for identities added to the agent, in minutes:
+
+`keychain --timeout {{minutes}}`
+
+- Enforce using gpg2 over gpg (for distributions like Ubuntu that have both)
```
Comment (CleanMachine1) [inline pages/common/keychain.md:26]:
```suggestion
- Use GPG2 rather than GPG
```

Comment (rpdelaney) [reply pages/common/keychain.md:26]: I would be open to just dropping this from the page if you think it's too high-level of a distinction for tldr.

Comment (CleanMachine1) [reply pages/common/keychain.md:26]: On my machine, GPG is a symbolic link to GPG2.

I don't know whether the command would be fine using the default method (without the --gpg2 flag), lets see what others think.

Comment (rpdelaney) [reply pages/common/keychain.md:26]: Are you using ubuntu? It's possible that the manpage for keychain is just out of date, and this feature is obsolete.

Comment (CleanMachine1) [reply pages/common/keychain.md:26]: Arch Linux

Comment (sbrl) [reply pages/common/keychain.md:26]: I'd assume that the only version of gpg used today is gpg2. I haven't heard of the original gpg1 being actually used at all?

Comment (CleanMachine1) [reply pages/common/keychain.md:26]: ok

**Review by CleanMachine1 — APPROVED** [review]

File: pages/common/keychain.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+> Re-use ssh-agent and/or gpg-agent between logins.
+> More information: <http://funtoo.org/Keychain>.
+
+- Check for a running ssh-agent, and start one if needed:
+
+`keychain`
+
+- Also check for gpg-agent:
+
+`keychain --agents "gpg,ssh"`
```
Comment (navarroaxel) [inline pages/common/keychain.md:12]: A token syntax is missing here
```suggestion
`keychain --agents "{{gpg,ssh}}"`
```

**Review by navarroaxel — APPROVED** [review]

**Pull request merged**